components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -0,0 +1,634 @@
+		end
+
+		if not subMatches[subGroupIndex] then
+			subMatches[subGroupIndex] = {scores = {0, 0}}
+		end
+
+		local winner = tonumber(map.winner)
+		if winner and subMatches[subGroupIndex].scores[winner] then
+			subMatches[subGroupIndex].scores[winner] = subMatches[subGroupIndex].scores[winner] + 1
+		end
+	end
+
+	Array.forEach(subMatches, function(subMatch, subMatchIndex)
+		-- get winner if the submatch is finished
+		-- submatch is finished if the next submatch has a score or if the complete match is finished
+		local nextSubMatch = subMatches[subMatchIndex + 1] or {scores = {0, 0}}
+
+		if not Logic.readBool(match.finished) and (nextSubMatch.scores[1] + nextSubMatch.scores[2] <= 0) then
+			return
+		end
+		if subMatch.scores[1] > subMatch.scores[2] then
+			subMatch.winner = 1
+		elseif subMatch.scores[2] > subMatch.scores[1] then
+			subMatch.winner = 2
+		end
+	end)
+
+	match.extradata.submatches = subMatches
+
+	return match
+end
+
+---@param match table
+---@return table
+function CustomMatchGroupInput._opponentInput(match)
+	Array.forEach(Array.range(1, MAX_NUM_OPPONENTS), FnUtil.curry(CustomMatchGroupInput.processOpponent, match))
+
+	return match
+end
+
+---@param match table
+---@param opponentIndex integer
+function CustomMatchGroupInput.processOpponent(match, opponentIndex)
+	local record = Json.parseIfString(match['opponent' .. opponentIndex])
+	assert(Opponent.isType(record.type), 'Unsupported Opponent Type')
+
+	local opponent = Opponent.readOpponentArgs(record) or Opponent.blank()
+
+	-- Convert byes to literals
+	if Opponent.isBye(opponent) then
+		opponent = {type = Opponent.literal, name = 'BYE'}
+	end
+
+	---@type number|string
+	local teamTemplateDate = match.timestamp
+	-- If date is default date, resolve using tournament dates instead
+	-- default date indicates that the match is missing a date
+	-- In order to get correct child team template, we will use an approximately date and not the default date
+	if teamTemplateDate == DateExt.defaultTimestamp then
+		teamTemplateDate = DateExt.getContextualDateOrNow()
+	end
+
+	Opponent.resolve(opponent, teamTemplateDate, {syncPlayer = true})
+
+	MatchGroupInput.mergeRecordWithOpponent(record, opponent)
+
+	--set initial opponent sumscore
+	record.sumscore = 0
+
+	if record.type == Opponent.team then
+		record.icon, record.icondark = CustomMatchGroupInput.getIcon(opponent.template)
+	end
+
+	match['opponent' .. opponentIndex] = record
+
+	if record.type == Opponent.team then
+		MatchGroupInput.readPlayersOfTeam(match, opponentIndex, record.name, {
+			resolveRedirect = true,
+			applyUnderScores = true,
+		})
+	else
+		Array.forEach(record.players or {}, function(player)
+			player.name = Page.pageifyLink(player.name)
+		end)
+	end
+
+	record.name = Page.pageifyLink(record.name)
+end
+
+---@param match table
+---@param mapIndex integer
+---@param subGroupIndex integer
+---@return table, integer
+function CustomMatchGroupInput._mapInput(match, mapIndex, subGroupIndex)
```

Comment:
Second return value seem to always be unmodified? It's always teh same as the 3rd input?